draft-ietf-jsonpath-base.md

```diff
@@ -1309,96 +1230,99 @@ Queries:
-slices, and filters in a single selector.
+The syntax and semantics of each __APPENDER__ are defined below.
 
-Note: The list selector was called "union selector" in
-{{JSONPath-orig}}, as it was intended to solve use cases addressed by
-the union selector in XPath.
-However, the term "union" has the connotation of a set operation that involves
-merging input sets while avoiding duplicates, so the concept was
-renamed into "list selector".
+### Child __APPENDER__
 
 #### Syntax
 {: unnumbered}
 
-The list selector is syntactically related to the
-`dot-selector`, `index-selector`, `slice-selector`, and the `filter-selector`.
-It contains two or more entries, separated by commas.
+The child __APPENDER__ consists of a non-empty, comma-delimited
+sequence of selectors enclosed in square brackets.
+
+Shorthand notations are also provided for when there is a single
+wildcard or name selector.
 
 ~~~~ abnf
-list-selector  = "[" S list-entry 1*(S "," S list-entry) S "]"
+child-APPENDER            = (child-longhand /
+                             dot-wildcard-shorthand /
+                             dot-member-name-shorthand)
 
-list-entry     =  ( quoted-member-name /
-                    element-index      /
-                    slice-index /
-                    filter
-                  )
-~~~~
+child-longhand            = "[" S selector 1*(S "," S selector) S "]"
 
-#### Semantics
-{: unnumbered}
+dot-wildcard-shorthand    = "." wildcard
 
-A list selector selects the nodes that are selected by
-the selector entries in the list and yields the concatenation of the
-lists (in the order of the selector entries) of nodes selected by the
-selector entries.
-Note that any node selected in more than one of the selector entries is kept
-as many times in the nodelist.
+dot-member-name-shorthand = "." dot-member-name
+dot-member-name           = name-first *name-char
+name-first                = ALPHA /
+                            "_"   /            ; _
+                            %x80-10FFFF        ; any non-ASCII Unicode character
+name-char                 = DIGIT / name-first
 
-To be valid, integer values in the `element-index` and `slice-index`
-components MUST be in the I-JSON {{-i-json}} range of exact values, see
-{{synsem-overview}}.
+DIGIT                     =  %x30-39              ; 0-9
+ALPHA                     =  %x41-5A / %x61-7A    ; A-Z / a-z
+~~~~
 
+The `dot-wildcard-shorthand` is shorthand for `[*]`.
 
-#### Examples
-{: unnumbered}
+A `dot-member-name-shorthand` of the form `.<member-name>` is shorthand for `['<member-name>']`.
 
-JSON:
+Member names containing characters other than allowed by
+`dot-member-name-shorthand` — such as space (U+0020), minus (U+002D), dot (U+002E)
+or escaped characters which appear in the [name selector semantics section](#name-semantics) —
+MUST NOT be used with the `dot-member-name-shorthand`.
+(In such cases, the `name-selector` syntax MUST be used instead.)
```

Comment:
This is a weird MUST, as it is a statement of fact ("if the key at hand does not fit the door, the key that does fit MUST be used"???).

Reply:
397f2befc51d47caf6403dccfc6c720468081ad1 deleted this.